lib/Conversion/ImportVerilog/ImportVerilog.cpp

```diff
@@ -236,8 +236,26 @@ LogicalResult ImportContext::prepareDriver(SourceMgr &sourceMgr) {
 /// Parse and elaborate the prepared source files, and populate the given MLIR
 /// `module` with corresponding operations.
 LogicalResult ImportContext::importVerilog(ModuleOp module) {
-  // This is where the Slang AST to CIRCT op conversion will go.
-  return success();
+  // Parse the input.
+  auto parseTimer = ts.nest("Verilog parser");
+  bool parseSuccess = driver.parseAllSources();
+  parseTimer.stop();
+
+  // Elaborate the input.
+  auto compileTimer = ts.nest("Verilog elaboration");
+  auto compilation = driver.createCompilation();
+  for (auto &diag : compilation->getAllDiagnostics())
+    driver.diagEngine.issue(diag);
+  if (!parseSuccess || driver.diagEngine.getNumErrors() > 0)
```

Comment:
Probably doesn't matter, but should the timer be stopped before returning failure?

Reply:
Is there a scope-based version that'll auto-stop on exit paths? I forget (and may not fit well here/style, just a thought)

Reply:
It does :slightly_smiling_face:. The `ts.nest()` call produces an RAII timer guard that you can either manually stop, or you just let it go out of scope to the same effect.